python-package/xgboost/data.py

```diff
@@ -909,9 +909,16 @@ def _transform_cudf_df(
     enable_categorical: bool,
 ) -> Tuple[ctypes.c_void_p, list, Optional[FeatureNames], Optional[FeatureTypes]]:
     try:
-        from cudf.api.types import is_categorical_dtype
+        from cudf.api.types import is_categorical_dtype, is_bool_dtype
     except ImportError:
         from cudf.utils.dtypes import is_categorical_dtype
+        from pandas.api.types import is_bool_dtype
+
+    # Work around https://github.com/dmlc/xgboost/issues/10181
+    if _is_cudf_ser(data) and is_bool_dtype(data.dtype):
+        data = data.astype(np.uint8)
+    else:
+        data = data.astype({col: np.uint8 for col in data.select_dtypes(include="bool")})
```

Comment:
Note. This workaround is only needed for cuDF. The handler for Pandas already converts all Boolean columns into float type.